sfcasts/menu-items.md

```diff
@@ -1,75 +1,35 @@
+
+Go back to the admin page, refresh... click on page and... perfect! This looks much better. But what about a link *back* to the admin page, like up here in the header? For that, let's go to `templates/base.html.twig`. Scroll down to the navbar. Here it is.
+
+There's nothing inside of it right now. I'm going to add `<li class="navbar-nav></li>` and a few other classes. And inside, add `<a>` that will `linkToPath`. And the, to link to the admin section, there's nothing special. Our `DashboardController` has a real route, and its name is `admin`, so we can just link to that admin route. Give our anchor a couple of classes to make it look good: `nav-link`... and I'll say "Admin" here.
+
+We really only want to render this link if we have `ROLE_ADMIN`. So `{% if isGranted('ROLE_ADMIN') %}` and then `{% endif %}` on the other side of this.
+
+Beautiful! Let's give that a try. Refresh... There's the link and... we're right back on our admin section.
+
+One other thing that the dashboard controls is this nice little user menu up here. It shows who you're logged in as, an avatar that doesn't work yet, and a "Sign out" link. In our system, users actually *do* have avatars on the front end. You can see this is an avatar for a user, and that's an avatar for a user up there. *But* EasyAdmin doesn't know that our users have avatars yet. We need to tell it.
+
+Back in our `DashboardController`... it doesn't matter where... I'm going to go to "Code Generate" or "cmd + N" on a Mac, go to "OverrideMethods", and select `configureUserMenu`. This has several methods on it. As you can see, we can add other menu items (we'll do that in second), set the avatar URL, and a few other things. I'll say `setAvatarUrl`.
+
+Then, say `$user->getAvatarUrl`. This is a custom method on my user class, so you'll notice I'm not getting auto complete on that. That's because PhpStorm doesn't know that this is our custom user class. So if you want to code defensively, you can add `if (!user instanceof User)`, and `throw new \exception( 'Wrong user.');`. That won't ever happen, but now if I retype this... `userAvatarUrl`... that fixes it. Now, when we refresh... perfect! We have our avatar up there!
+
+The last thing I would like to have is a little link down here that goes to my profile. Earlier, we saw that another method you can call is `setMenuItems`, where you can pass it in array of menu items. These items are the same ones we've been building down here. So we can say, for example, `MenuItem::linkToUrl` with `'MyProfile'`... and give this some icons.
+
+Then I'll say `$this->generateUrl`... and the name to my route for my profile page is `app_profile_show`. That's it! I don't need any other argument. Refresh and... that works! Click on "My Profile" and... that works too!
```

Comment:
This should be part of the previous paragraph, as we're right in the middle of finishing a line of code.